Documentation/CodingGuidelines/CglPhp/PhpArchitecture/WorkingWithExceptions.rst

```diff
@@ -95,6 +95,35 @@ a meaningful message in is important especially if top-level exceptions
 are thrown. A developer receiving this exception should get all useful
 data that can help to debug and mitigate the issue.
 
+Example
+
+    try {
+        if ($theFile == '') {
+            throw new \Exception ('The import file has not been found.');
+        }
+        if ($pid == 0) {
+            throw new \Exception ('The page "' . $pid . '" cannot be accessed.');
+        }
+
+        $absFilename = \TYPO3\CMS\Core\Utility\GeneralUtility::getFileAbsFileName($theFile);
+
+        if (is_file($absFilename)) {
+            $iFile = fopen($absFilename, 'rb');
+        } else {
+            throw new \Exception ('File "' . $absFilename . '" not found.');
+        }
+
+        if ($iFile == null) {
+            throw new \Exception ('File "' . $absFilename . '" cannot be read.');
+        }
+    } // try
+
+    catch (\Exception $e) {
+        throw new \TYPO3\CMS\Core\Exception($e->getMessage(), exec('date +%s'));
```

Comment:
The suggestion about the exception code is meant to be executed every time a new exception is thrown in code. The important bit is that the exception code never ever changes again. Thus executing a shell command to generate an exception code does not make sense, instead a static one must be inserted here, e.g. `1548144780`.

Reply:
> Since this is the UNIX timestamp the chances are very low that you hit exactly the same code as other extension authors.

But you suggested to use a static code. So how should this be solved then?

Reply:
See my latest change.

Reply:
Thanks. I never understood this even after reading this documentation or after Google searches.